### PR TITLE
Add metadata

### PR DESCRIPTION
**Pull request by ericpre**
See #1222.

For now, I updated only the protochips io plugin, but we are all ok with the format suggested in #1222, then I can try to go through the other plugins to set the quantity, date and time properly.
I would like to have #1222 fixed to move on with #546.

With date and time saved as string in ISO 8601, this is very easy to concert it to numpy.datetime64:
`numpy.datetime64("%sT%s"%(date, time))` where date and time come from the metadata tree of a signal `s.metadata.General.date` and `s.metadata.General.time`, respectively.

### CODE REVIEW

File: doc/user_guide/metadata_structure.rst [hunk 1]
```diff
@@ -88,15 +92,30 @@ original_filename
     original file.
 
 time
-    type: datetime.time
+    type: Str
 
-    The acquistion or creation time.
+    The acquistion or creation time in ISO 8601 time format.
```
Comment (francisco-dlp) [inline doc/user_guide/metadata_structure.rst]: inherited typo: acquisition

File: doc/user_guide/metadata_structure.rst [hunk 2]
```diff
@@ -88,15 +92,30 @@ original_filename
...
 
 date
-    type: datetime.time
+    type: Str
+
+    The acquistion or creation date in ISO 8601 date format
```
Comment (francisco-dlp) [inline doc/user_guide/metadata_structure.rst]: As above

File: doc/user_guide/metadata_structure.rst [hunk 3]
```diff
@@ -88,15 +92,30 @@ original_filename
...
+
+
+authors
+    type: Str
+
+    The authors who generated the files.
```
Comment (francisco-dlp) [inline doc/user_guide/metadata_structure.rst]: The authors of the data?

File: doc/user_guide/metadata_structure.rst [hunk 4]
```diff
@@ -88,15 +92,30 @@ original_filename
...
 
-    The acquistion or creation date.
+doi
+    type: Str
+
+    Digital object identifier.
```
Comment (francisco-dlp) [inline doc/user_guide/metadata_structure.rst]: Digital object identified of the data?

File: doc/user_guide/metadata_structure.rst [hunk 5]
```diff
@@ -300,6 +319,11 @@ record_by
     One of 'spectrum' or 'image'. It describes how the data is stored in memory.
     If 'spectrum' the spectral data is stored in the faster index.
 
+quantity
+    type: Str
+    
+    The name of the quantity of the "intensity axis" with the units in brackets if required.
```
Comment (francisco-dlp) [inline doc/user_guide/metadata_structure.rst]: round brackets?

**Comment by francisco-dlp** [comment]
Looks good! If you could go through the other plugins that would be great.

You could also add the current version of the specification, which should be the same as the version of the hdf5 reader. Currently it is 2.1. As you're adding features but not breaking anything, the new version should be 2.2.

**Comment by ericpre** [comment]
Thanks for reviewing!
I don't really understand the following:
> You could also add the current version of the specification, which should be the same as the version of the hdf5 reader. Currently it is 2.1. As you're adding features but not breaking anything, the new version should be 2.2.

Do you mean that the changes in metadata specificaitons should also be documentated as changes in the specifications of the HyperSpy hdf5 format? At the beginning of the hdf5 io plugin?

**Comment by francisco-dlp** [comment]
You could indeed add a sentence there. Ideally we would fully document the format in a single document containing the current metadata specification + all the details of how we store stuff in the hdf5 file. This document should probably live in a separate repository. However, at present, things are far from perfect in this regard...

**Comment by Hovden** [comment]
+1 on 8601 date, time metadata

Just noting that image stacks might have several timestamps. For example, this metadata was taken from FEI .ser dataset using hyperspy:

 Time = array([1467843134, 1467843139, 1467843144, 1467843150, 1467843155,
       1467 ...     1467843212, 1467843217, 1467843223, 1467843228, 1467843233], dtype=uint32)

**Comment by ericpre** [comment]
Thanks for pointing this out! I have been through updating the FEI reader, considered only the date/time information of the emi file and neglect the one of the ser file, which was a bit lazy from my side, because in principle, we should be able to open ser files as good as we do for emi anyway...

Actually, the fact there is array of time stamp in the ser header of stack images is interesting because the time array could be use for the `navigation_axis` of the `axes_manager`, which I think for now is undefined. However, I would expect the axis not to be always linear... so that's one more for #438!

**Comment by francisco-dlp** [comment]
Good points @Hovden  and @ericpre. Indeed we were tempted to set the time array in the ser file as the nav axis but we didn't because of #438 as @ericpre mentions and also because the accuracy is just 1s.

**Comment by ericpre** [comment]
Resubmitted to RELEASE_next_patch #1240.

**Pull request closed without merging**